src/common/inference/service.rs

```diff
@@ -180,24 +205,72 @@ impl InferenceService {
         inference_type: InferenceType,
         inference_token: InferenceToken,
     ) -> Result<InferenceResponse, StorageError> {
+        let (bm25_inference_inputs, inference_inputs): (Vec<_>, Vec<_>) = inference_inputs
```

Comment:
I would prefer to keep `infer` function agnostic to specialized implementations.

Instead of splitting inputs into `bm25_inference_inputs, inference_inputs` I propose to split into `local_inference_inputs, remote_inference_inputs`.

Also, I propose to move all bm25 config parsing and inference into a separate file.